src/sidebar/services/frame-sync.ts

```diff
@@ -267,6 +282,33 @@ export class FrameSyncService {
           this._updateAnchorStatus(anchorImmediately, 'anchored');
         }
 
+        // For added annotations that still need to anchor, re-check them after
+        // ANCHORING_TIMEOUT has elapsed. Any added annotations not anchored
+        // by that time should be marked as having an anchor timeout.
+        const anchoringIds = added
+          .filter(ann => isWaitingToAnchor(ann))
+          .map(ann => ann.id);
+
+        const ANCHORING_TIMEOUT = 500;
+
+        if (anchoringIds.length > 0) {
```

Comment:
The containing method is getting quite long, so it might be wise to extract the body of this conditional into a separate method.

Reply:
Yeah, I was on the fence about that. The rest of the contents of the method were fairly "unrolled," so I rolled with that. Happy to extract.